src/Analyzer/JoinNode.cpp

```diff
@@ -113,4 +118,17 @@ ASTPtr JoinNode::toASTImpl() const
     return tables_in_select_query_ast;
 }
 
+void JoinNode::crossToInner(const QueryTreeNodePtr & join_expression_)
+{
+    if (kind != JoinKind::Cross && kind != JoinKind::Comma)
+        throw Exception(ErrorCodes::LOGICAL_ERROR, "Cannot rewrite join {} to inner join, expected cross", toString(kind));
```

Comment:
Join kind `toString` method returns join kind string representation in uppercase, we need to do the same in exception message.